release/nightly_tests/dataset/benchmark.py

```diff
@@ -0,0 +1,55 @@
+import gc
+import json
+import os
+import time
+from typing import Callable
+
+from ray.data.dataset import Dataset
+
+
+class Benchmark:
```

Comment:
What's the scope of the Benchmark? IIUC it's a benchmark for Dataset transformations, if so maybe make it more clear about that. Also good to mention that if it's applicable for both local and distribute benchmarking.

Reply:
Ideally the scope of `Benchmark` should cover all of data-related benchmark (dataset, dataset pipeline, transform, action, etc), there's no restrict to be used only for dataset transformation. It works for both local and distribute benchmarking. let me add more documentation.

Reply:
@jianoaix - updated.

Reply:
If `fn` is Dataset-to-Dataset mapping, it's basically a transform? Like iter_batches(), min/max etc are not covered.

Reply:
oh, it's just for easy to retrieve the statistics by returning another Dataset. You can do arbitrary logic inside benchmark:

```
def fn(input_ds):
    input_ds.iter_batches(...)
    input_ds.min()
    input_ds.max()
    return the_ds_you_care_for_stats
```

Reply:
also just to add - the parameter to `fn` can be everything, so we are not bound to pass a Dataset.

Reply:
I'd add a comment about what's expected to return for fn.

Reply:
@jianoaix - sure, added. Also `run(fn: Callable[..., Dataset])` has function's expected return type.